include/extractor/edge_based_node.hpp

```diff
@@ -13,6 +13,7 @@ struct EdgeBasedNode
     GeometryID geometry_id;
     ComponentID component_id;
     AnnotationID annotation_id;
+    bool segregated;
```

Comment:
we should see if we can use a single bit from the annotation-id for this, to not increase the size of an edge-based node.

Reply:
Sounds good. At this time we have about 8000000 different annotations for the whole world. Should I do it or leave as is for now?
Also, I have thoughts to add segregated flag into Annotation struct, but the logic will be more complex.

Reply:
The `segregated` bit can be merged with `AnnotationID` to avoid wasting space of  31 padding bits.